frontend/app/src/composables/history/filter-paginate.ts

```diff
@@ -142,15 +136,11 @@ export const useHistoryPaginationFilter = <T extends Object, U, V>(
     set(paginationOptions, newOptions);
   };
 
-  const setFilter = (newFilter: UnwrapRef<typeof filters>) => {
+  const setFilter = (newFilter: UnwrapRef<Ref<W>>) => {
     set(userAction, true);
     updateFilter(newFilter);
   };
 
-  onBeforeMount(() => {
-    applyRouteFilter();
-  });
```

Comment:
What is the problem with having it inside the composable?

Reply:
it is causing an issue in the test environment...

on a second thought, I could mock the `onBeforeMount` composable.

Reply:
Yes, if it is only for testing then maybe we can mock it. 
Because just copying it in every component would be really inconvenient